src/overlays/actors/ovl_En_Fall2/z_en_fall2.c

```diff
@@ -28,26 +33,137 @@ const ActorInit En_Fall2_InitVars = {
+    perVertexAlphaTable[3] = (s8)(155.0f * arg0);
+    perVertexAlphaTable[4] = (s8)(255.0f * arg0);
+    perVertexAlphaTable[5] = (s8)(205.0f * arg0);
+
+    for (i = 0; i < ARRAY_COUNT(sAlphaTableIndices); i++, vertex++) {
+        vertex->v.cn[3] = perVertexAlphaTable[sAlphaTableIndices[i]];
+    }
+}
+
+void EnFall2_DoNothing(EnFall2* this, PlayState* play) {
+}
+
+void func_80C1B8B4(EnFall2* this) {
+    this->actor.draw = EnFall2_Draw;
+    if (this->csActorAction == 1) {
+        Actor_SetScale(&this->actor, 0.02f);
+    }
+}
+
+void func_80C1B8F0(EnFall2* this) {
+    switch (this->csActorAction) {
+        case 1:
+            if (this->alphaLevel < 1.0f) {
+                this->alphaLevel += 1.0f / 30.0f;
+            } else {
+                this->alphaLevel = 1.0f;
+            }
+            break;
+        case 2:
+            if (this->alphaLevel > 0.0f) {
+                this->alphaLevel -= 0.1f;
+                if (this->alphaLevel < 0.0f) {
+                    this->alphaLevel = 0.0f;
+                }
+            }
+            if (this->actor.scale.x > 0.0f) {
+                this->actor.scale.x -= 20.0f * 0.0001f;
+                if (this->actor.scale.x < 0.0f) {
+                    this->actor.scale.x = 0.0f;
+                }
+                this->actor.scale.z = this->actor.scale.x;
+            }
+    }
+}
+
+void func_80C1B9D4(EnFall2* this, PlayState* play) {
+    func_80183DE0(&this->skeletonInfo);
+    if (Cutscene_CheckActorAction(play, this->csActionIndex)) {
+        Cutscene_ActorTranslateAndYaw(&this->actor, play, Cutscene_GetActorActionIndex(play, this->csActionIndex));
+        if (this->csActorAction !=
+            play->csCtx.actorActions[Cutscene_GetActorActionIndex(play, this->csActionIndex)]->action) {
+            this->csActorAction =
+                play->csCtx.actorActions[Cutscene_GetActorActionIndex(play, this->csActionIndex)]->action;
+            func_80C1B8B4(this);
+        }
+        func_80C1B8F0(this);
+        if (this->alphaLevel > 0.0f) {
+            func_800B9010(&this->actor, NA_SE_EV_MOON_LIGHT_PILLAR - SFX_FLAG);
+        }
+    } else {
+        this->actor.draw = NULL;
+    }
+    func_80C1B724(this->alphaLevel);
+}
+
+void EnFall2_Update(Actor* thisx, PlayState* play) {
+    EnFall2* this = THIS;
+
+    this->actionFunc(this, play);
+}
+
+void EnFall2_Draw(Actor* thisx, PlayState* play) {
+    s32 pad;
+    EnFall2* this = THIS;
+    Mtx* mtx;
+
+    if (!(this->alphaLevel <= 0.0f)) {
+        func_8012C2DC(play->state.gfxCtx);
+        AnimatedMat_DrawXlu(play, Lib_SegmentedToVirtual(object_fall2_Matanimheader_008840));
+
+        mtx = GRAPH_ALLOC(play->state.gfxCtx, ALIGN16(this->skeletonInfo.unk_18->unk_1 * sizeof(Mtx)));
+
+        if (mtx != NULL) {
+            func_8012C2DC(play->state.gfxCtx);
+            Matrix_RotateYS((s16)(Camera_GetCamDirYaw(play->cameraPtrs[play->activeCamera]) + 0x8000), MTXMODE_APPLY);
```

Comment:
```suggestion
            Matrix_RotateYS((s16)(Camera_GetCamDirYaw(GET_ACTIVE_CAM(play)) + 0x8000), MTXMODE_APPLY);
```
There's a macro that you should use here!